### PR TITLE
Remove trailing && if present

### PR DESCRIPTION
**Pull request by AdamBrousseau**
Issue infra 7211

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
We won't hit this code path in a public build, other than the `if` condition.

**Comment by AdamBrousseau** [comment]
jenkins test sanity amac jdk11

**Comment by AdamBrousseau** [comment]
jenkins test sanity amac jdk11

File: buildenv/jenkins/common/variables-functions.groovy [hunk 1]
```diff
@@ -755,6 +755,9 @@ def set_test_misc() {
         // Only add extra test labels if the user has not specified a specific TEST_NODE
         TESTS.each { id, target ->
             target['extraTestLabels'] = buildspec.getVectorField("extra_test_labels", SDK_VERSION).join('&&') ?: ''
+            if (target['extraTestLabels'].endsWith('&&')){
+                target['extraTestLabels'] = target['extraTestLabels'].substring(0, s.length() - 2)
```
[the remark this answers is not in the document]
Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions.groovy:759]: bad copy paste

**Comment by AdamBrousseau** [comment]
Pushed update.
Test build from last run
https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/4052/

**Review by pshipton — APPROVED** [review]

**Pull request merged**